emuu-client/src/components/CreatorsPage/CreatorsPage.js

```diff
@@ -1,83 +1,308 @@
 import {
-  getDoc,
+  getFirestore,
+  collection,
   getDocs,
+  updateDoc,
   setDoc,
   doc,
-  collection,
+  getDoc,
   query,
-  where,
-  onSnapshot,
-  increment,
-  updateDoc,
 } from "firebase/firestore";
-import HeaderPostLogin from "../NavbarPostLogin/HeaderPostLogin.js";
+import { db } from "../../Firebase.js";
 
 //Function to display creator page
-function Creator({ setVideo, video, setUserProfile }) {
-  const [creatorName, setCreatorName] = useState("Temp");
-  const docRef = doc(db, "Users", creatorName);
+const Creator = ({ setVideo, video }) => {
+  const history = useHistory();
+  const location = useLocation();
+  console.log(location, "location");
+  const creatorsData = JSON.parse(localStorage.getItem("creatorsData"))[0];
+  const subscriberVideos = JSON.parse(
+    localStorage.getItem("creatorsDataVideos")
+  );
+  const [autocompleteState, setAutocompleteState] = useState({});
+  const [searchInput, setSearchInput] = useState("");
+  const [count, setCount] = useState(0);
+  const [subscriberActionCount, setSubsciberActionCount] = useState(0);
+  const [updatedSubscribersList, setUpdateSubscribersList] = useState([]);
 
-  useEffect(async () => {
-    if (video) {
-      localStorage.setItem("video", JSON.stringify(video));
+  const firebaseData = JSON.parse(localStorage.getItem("firebase-data"));
+
+  const autocomplete = useMemo(
+    () =>
+      createAutocomplete({
+        onStateChange({ state }) {
+          setAutocompleteState(state);
+          setSearchInput(state.query);
+          if (count === 0) {
+            setCount((count) => count + 1);
+          }
+        },
+        getSources() {
+          return [
+            {
+              sourceId: "pages-source",
+              getItemInputValue({ item }) {
+                // search item
+                return item.query;
+              },
+              getItems({ query }) {
+                // takes your search input and checks if anything that matches it exists in your dataset
+                if (!query) {
+                  return firebaseData;
+                }
+                return firebaseData.filter(
+                  (item) =>
+                    item.VideoTitle?.toLowerCase().includes(
+                      query.toLowerCase()
+                    ) ||
+                    item.Username?.toLowerCase().includes(
+                      query.toLocaleLowerCase()
+                    )
+                );
+              },
+              templates: {
+                item({ item }) {
+                  return item.VideoTitle || item.Username;
+                },
+              },
+            },
+          ];
+        },
+      }),
+    [count]
+  );
+
+  const dataSet = autocompleteState?.collections?.[0]?.items;
+  const searchResultsVideosArr = dataSet?.filter(
+    (obj) => obj.hasOwnProperty("VideoUrl") && obj.hasOwnProperty("Username")
+  );
+  const searchResultsUsersArr = dataSet?.filter(
```

Comment:
If you're using search function on every page, why not just create one function and call it on the pages? This leads to more cleaner and efficient code.